include/seqan3/range/view/istreambuf.hpp

```diff
@@ -0,0 +1,248 @@
+
+namespace seqan3::detail
+{
+
+// ============================================================================
+//  fast_istreambuf_iterator
+// ============================================================================
+
+//!\brief Functionally the same as std::basic_streambuf<char_t, traits_t_>, but exposes protected members as public.
+template <typename char_t, typename traits_t = std::char_traits<char_t>>
+struct stream_buffer_exposer : public std::basic_streambuf<char_t, traits_t>
+{
+    //!\brief The actual stream type.
+    using base_t = std::basic_streambuf<char_t, traits_t>;
+
+    // Expose protected members:
+    using base_t::eback;
+    using base_t::gptr;
+    using base_t::egptr;
+    using base_t::gbump;
+    using base_t::underflow;
+
+    using base_t::pbase;
+    using base_t::pptr;
+    using base_t::epptr;
+    using base_t::pbump;
+    using base_t::overflow;
+};
+
+/*!\brief Functionally the same as std::istreambuf_iterator, but faster.
+ * \tparam char_t       The stream's character type.
+ * \tparam traits_t_    The stream's traits type.
+ *
+ * \details
+ *
+ * Performs less virtual function calls than std::istreambuf_iterator.
+ *
+ * \todo Make this move-only after input iterators are allowed to be move-only.
+ *
+ */
+template <typename char_t, typename traits_t = std::char_traits<char_t>>
+class fast_istreambuf_iterator
+{
+private:
+    //!\brief Down-cast pointer to the stream-buffer.
+    stream_buffer_exposer<char_t, traits_t> * stream_buf = nullptr;
+
+public:
+    /*!\name Associated types
+     * \{
+     */
+    using difference_type   = ptrdiff_t;               //!< Defaults to ptrdiff_t.
+    using value_type        = char_t;                  //!< The char type of the stream.
+    using reference         = char_t;                  //!< The char type of the stream.
+    using pointer           = void;                    //!< Has no pointer type.
+    using iterator_category = std::input_iterator_tag; //!< Pure input iterator.
+    //!\}
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    fast_istreambuf_iterator()                                              noexcept = default; //!< Defaulted.
+    fast_istreambuf_iterator(fast_istreambuf_iterator const &)              noexcept = default; //!< Defaulted.
+    fast_istreambuf_iterator(fast_istreambuf_iterator &&)                   noexcept = default; //!< Defaulted.
+    fast_istreambuf_iterator & operator=(fast_istreambuf_iterator const &)  noexcept = default; //!< Defaulted.
+    fast_istreambuf_iterator & operator=(fast_istreambuf_iterator &&)       noexcept = default; //!< Defaulted.
+    ~fast_istreambuf_iterator()                                             noexcept = default; //!< Defaulted.
+
+    //!\brief Construct from a stream buffer.
+    explicit fast_istreambuf_iterator(std::basic_streambuf<char_t, traits_t> & ibuf) :
+        stream_buf{reinterpret_cast<stream_buffer_exposer<char_t, traits_t> *>(&ibuf)}
```

Comment:
not sure this valid. Have you verified that this does not trigger warnings with sanitizers enabled.
You trying to tell the compiler that this class is just another way of accessing the bit representation, but the members are all protected and could be overloaded and interleaved with other members in the derived class of basic_streambuf you are getting here. That's why I had to wrap it in SeqAn2 in the first place.

Reply:
Well, we are just storing a pointer, or not?

Reply:
The following example at least does not produce sanitzier warnings:
```cpp

struct S
{
protected:
   void foo() {};
};

struct T : S
{
    using S::foo;
};

int main()
{
    S s;
    T * t = reinterpret_cast<T*>(&s);
}
```